src/type-checker.cc

```diff
@@ -706,13 +740,24 @@ Result TypeChecker::OnReturn() {
 
 Result TypeChecker::OnSelect() {
   Result result = Result::Ok;
-  Type type = Type::Any;
+  Type type1 = Type::Any;
+  Type type2 = Type::Any;
+  Type result_type = Type::Any;
   result |= PeekAndCheckType(0, Type::I32);
-  result |= PeekType(1, &type);
-  result |= PeekAndCheckType(2, type);
-  PrintStackIfFailed(result, "select", type, type, Type::I32);
+  result |= PeekType(1, &type1);
+  result |= PeekType(2, &type2);
+  // If both types are reftypes then check for common subtype
```

Comment:
should be "check for common supertype"?

Reply:
I actually think that none of this will actually be needed.. i just saw that reference-types proposal upstream decided against requiring this type of type deduction.

Reply:
This code is now gone with the latest testsuite update.